### PR TITLE
[fix]: [CCM-8935]: Separate includeOthers data with BMOthers data in chart

### PR DESCRIPTION
**Pull request by iamanmolm**
## Describe your changes
We have written migration script, where we fetch all the business mapping which has label for Unallocated Cost as "Others" or "Unallocated". And then updating such business mapping Unallocated Cost label to "Unattributed".

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39337)

### CODE REVIEW

**Comment by USB-05** [comment]
trigger sspoc

**Comment by USB-05** [comment]
trigger bnpoc

**Comment by USB-05** [comment]
trigger bnpoc

**Comment by USB-05** [comment]
trigger bnpoc

**Comment by USB-05** [comment]
trigger rcpoc

**Comment by USB-05** [comment]
trigger mcpoc

**Comment by USB-05** [comment]
trigger allpoc

**Comment by USB-05** [comment]
trigger allpoc

**Comment by iamanmolm** [comment]
trigger codebasehash

**Comment by iamanmolm** [comment]
trigger codebasehash

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BudgetYearlyAddBreakdownToggleMigration.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.ccm.migration;
+
+public class BudgetYearlyAddBreakdownToggleMigration {}
```
Comment (sdsingh07) [inline ce-nextgen/service/src/main/java/io/harness/ccm/migration/BudgetYearlyAddBreakdownToggleMigration.java:10]: Remove this file.

Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/migration/BudgetYearlyAddBreakdownToggleMigration.java]: Done

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  @Override
+  public void migrate() {
+    try {
+      log.info(
+          "Starting migration (updates) of all Business Mapping Unallocated Cost Label from 'Others' to 'Unattributed'");
+      final List<BusinessMapping> businessMappingList =
+          hPersistence.createQuery(BusinessMapping.class, excludeAuthority).asList();
+      for (final BusinessMapping businessMapping : businessMappingList) {
+        if (businessMapping.getUnallocatedCost() != null && businessMapping.getUnallocatedCost().getLabel() != null
+            && businessMapping.getUnallocatedCost().getLabel().equals(OTHERS)) {
```
Comment (sdsingh07) [inline ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java:41]: Can we check whether in the prod, do we also have Unallocated label. If yes, then we need to migrate those cost categories as well.

Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java]: Done

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import io.harness.persistence.HPersistence;
+
+import com.google.inject.Inject;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@OwnedBy(HarnessTeam.CE)
+public class BusinessMappingUnallocatedLabelMigration implements NGMigration {
+  private static final String OTHERS = "Others";
```
Comment (sdsingh07) [inline ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java:28]: We can remove this constant.

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Override
+  public void migrate() {
+    try {
+      log.info(
+          "Starting migration (updates) of all Business Mapping Unallocated Cost Label from 'Others' to 'Unattributed'");
+      final List<BusinessMapping> businessMappingList =
+          hPersistence.createQuery(BusinessMapping.class, excludeAuthority).asList();
+      for (final BusinessMapping businessMapping : businessMappingList) {
+        if (businessMappingService.isValidBusinessMappingUnallocatedCostLabel(businessMapping)) {
```
Comment (sdsingh07) [inline ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java:42]: If it's not valid, then migrate it.
`if (!businessMappingService.isValidBusinessMappingUnallocatedCostLabel(businessMapping))`

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java (hunk 3 above)
Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java:28]: Done

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java (hunk 4 above)
Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/migration/BusinessMappingUnallocatedLabelMigration.java:42]: Already resolved.

**Review by sdsingh07 — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, 4 of 5 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39337#-:-NFr9WUMDu8APdgeLMH-:bnfp4nl)* status: :shipit: complete! ✓

**Comment by iamanmolm** [comment]
trigger ti1

**Comment by iamanmolm** [comment]
trigger compile, ti0, ti4

**Comment by iamanmolm** [comment]
trigger codebasehash

**Comment by iamanmolm** [comment]
trigger compile

**Review by sdsingh07 — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39337#-:-NFw3_kKA9Sx6U5qILMN:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**